pkg/tfbridge/provider.go

```diff
@@ -555,7 +557,18 @@ func (p *Provider) Configure(ctx context.Context,
 // https://github.com/hashicorp/terraform/blob/7f5ffbfe9027c34c4ce1062a42b6e8d80b5504e0/helper/schema/schema.go#L1356
 var requiredFieldRegex = regexp.MustCompile("\"(.*?)\": required field is not set")
 
-func (p *Provider) formatFailureReason(res Resource, reason string) string {
+func (p *Provider) formatFailureReason(tokenType tokens.Type, res Resource, err error) string {
+	reason := err.Error()
+	var attributePath string
+	var d *diagnostics.Error
+	if errors.As(err, &d) {
+		path := d.AttributePath
+		if len(path) > 0 {
+			attrPath := pathToAttributePath(path, tokenType, res)
+			attributePath = strings.Join(attrPath, "")
+		}
+	}
+
```

Comment:
Does this work for all of the various shims, or will this only work for the v2 shim?

Reply:
> Does this work for all of the various shims, or will this only work for the v2 shim?

I believe it should but I focussed on the v2 shim for now. I will verify before marking PR ready for review.

Reply:
I was wrong. I added support for tf5 but it appears v1's terraform-plugin-sdk doesn't actually return attribute information. It does appear though that it provides some useful information in the returned error but short of parsing the error messages, don't think this approach extends to the v1 shim.